ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -531,23 +532,44 @@ impl FpuState {
             unsafe { _fxrstor64(mem_addr) };
         }
 
-        self.is_valid.store(false, Relaxed);
-
         debug!("Restore FPU state");
     }
 
-    /// Clears the state of the instance.
-    ///
-    /// This method does not reset the underlying buffer that contains the
-    /// FPU state; it only marks the buffer __invalid__.
-    pub fn clear(&self) {
-        self.is_valid.store(false, Relaxed);
+    /// Saves CPU's current FPU state into signal context area.
+    pub fn save_to_signal_area(&self, xsave_area: &mut XSaveArea) {
+        self.save();
+        *xsave_area = *self.state_area;
+
+        // reset state
```

Comment:
You don't need to reset the state of `self`, right? You only want to restore the FPU state to its default. How about directly using `default_state` instead of copying its contents to `self`? This way, you can avoid the unnecessary copy.

```rust
let default_state = XSaveArea::default();
let mem_addr = &default_state as *const u8;
if CPU_FEATURES.get().unwrap().has_xsave() {
    unsafe { _xsave64(mem_addr, XFEATURE_MASK_USER_RESTORE) };
} else {
    unsafe { _fxsave64(mem_addr) };
}
```